taichi/program/state_flow_graph.cpp

```diff
@@ -727,6 +727,10 @@ std::unordered_set<int> StateFlowGraph::fuse_range(int begin, int end) {
   for (int i = 0; i < n; i++) {
     fused[i] = nodes[i]->rec.empty();
   }
+  auto get_initial_remaining_fuses = [n, this]() {
+    const auto configured_max = program_->config.async_max_fuse_per_task;
+    return configured_max > 0 ? configured_max : n;
+  };
```

Comment:
I suggest appending `_per_task` to this variable name to reduce confusion.

BTW what's the benefit to write it as a lambda expression instead of a const int (since `n` never changes in this function)?

Reply:
>  what's the benefit to write it as a lambda expression

I started writing this with the intention of passing in the upper bound, then noticed that we can just use `n`, so no real benefit.. Fixed